Portal/vip-api/src/main/java/fr/insalyon/creatis/vip/api/business/ExecutionBusiness.java

```diff
@@ -31,25 +31,329 @@
+        } catch (BusinessException ex) {
+            throw new ApiException(ex);
+        }
+    }
+
+    public ExecutionStatus playExecution(String executionId) throws ApiException {
+        try {
+            // Execution cannot be "played" (i.e. started) because it was already started in initExecution method.
+            // So we just return the execution status.
+            WorkflowBusiness wb = new WorkflowBusiness();
+            getWarnings().add("In VIP, playExecution only returns the status of the execution.");
+            return VIPtoCarminStatus(wb.getSimulation(executionId).getStatus());
+        } catch (BusinessException ex) {
+            throw new ApiException(ex);
+        }
+    }
+
+    public void killExecution(String executionId) throws ApiException {
+        WorkflowBusiness wb = new WorkflowBusiness();
+        try {
+            wb.kill(executionId);
+        } catch (BusinessException ex) {
+            throw new ApiException(ex);
+        }
+    }
+
+    public void deleteExecution(String executionId, Boolean deleteFiles) throws ApiException {
+        checkIfUserCanAccessExecution(executionId);
+        WorkflowBusiness wb = new WorkflowBusiness();
+        try {
+            Simulation s = wb.getSimulation(executionId);
+            if (s.getStatus() != SimulationStatus.Completed && s.getStatus() != SimulationStatus.Killed) {
+                throw new ApiException("Cannot delete execution " + executionId + " because status is " + s.getStatus().toString());
+            }
+            // Note: this won't delete the intermediate files in case the execution was run locally, which violates the spec.
+            // Purge should be called in that case but purge also violates the spec.
+            wb.clean(executionId, getUser().getEmail(), deleteFiles);
+        } catch (BusinessException ex) {
+            throw new ApiException(ex);
+        }
+    }
+
+    ;
+    public String[] getExecutionResults(String executionId, String protocol) throws ApiException {
+        try {
+            if(protocol == null)
+                protocol = "https";
+
+            if (!protocol.equals("https") && !protocol.equals("http")) {
+                throw new ApiException("Unsupported protocol: " + protocol);
+            }
+
+            ArrayList<String> urls = new ArrayList<>();
+
+            TransferPoolBusiness tpb = new TransferPoolBusiness();
+            WorkflowBusiness wb = new WorkflowBusiness();
+            List<InOutData> outputs = wb.getOutputData(executionId, getUser().getFolder());
+            for (InOutData output : outputs) {
+                
+                String operationId = tpb.downloadFile(getUser(), output.getPath());
+                
+                String url = getRequest().getRequestURL() + "/../fr.insalyon.creatis.vip.portal.Main/filedownloadservice?operationid=" + operationId; 
+                URL u = new URL(url); // just to check that it is a well-formed URL
```

Comment:
Yes, because the method returns a string. The URL is built just to check that it is well formed. I could also return u.toString(), but this is equivalent.